forge/forge/components/user_interaction/user_interaction.py

```diff
@@ -4,16 +4,15 @@
 
 from forge.agent.protocols import CommandProvider
 from forge.command import Command, command
-from forge.config.config import Config
 from forge.models.json_schema import JSONSchema
 from forge.utils.const import ASK_COMMAND
 
 
 class UserInteractionComponent(CommandProvider):
     """Provides commands to interact with the user."""
 
-    def __init__(self, config: Config):
-        self._enabled = not config.noninteractive_mode
+    def __init__(self, noninteractive_mode: bool):
+        self._enabled = not noninteractive_mode
```

Comment:
Why instantiate a component if it is disabled through its entire lifecycle?

Reply:
Removed and updated `Agent`.